dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -188,6 +188,63 @@ class AppEngineCocoonService implements CocoonService {
     }
   }
 
+  @override
+  Future<CocoonResponse<List<TreeStatusChange>>> fetchTreeStatusChanges({
+    required String repo,
+  }) async {
+    final getTreeStatusChangesUrl = apiEndpoint(
+      '/api/public/get-tree-status-changes',
+      queryParameters: {'repo': repo},
+    );
+
+    final response = await _client.get(getTreeStatusChangesUrl);
+
+    if (response.statusCode != HttpStatus.ok) {
+      return CocoonResponse.error(
+        '/api/public/get-tree-status-changes returned ${response.statusCode}',
+        statusCode: response.statusCode,
+      );
+    }
+
+    try {
+      final jsonResponse = jsonDecode(response.body) as List<Object?>;
+      final changes = <TreeStatusChange>[];
+      for (final jsonChange in jsonResponse.cast<Map<String, Object?>>()) {
+        changes.add(TreeStatusChange.fromJson(jsonChange));
+      }
+      return CocoonResponse.data(changes);
+    } catch (error) {
+      return CocoonResponse.error(
+        error.toString(),
+        statusCode: response.statusCode,
+      );
+    }
+  }
+
+  @override
+  Future<CocoonResponse<void>> updateTreeStatus({
+    required String repo,
+    required TreeStatus status,
+    String? reason,
+  }) async {
+    final updateTreeStatusUrl = apiEndpoint('/api/public/update-tree-status');
+    final response = await _client.post(
+      updateTreeStatusUrl,
+      body: jsonEncode({
+        'repo': repo,
+        'status': status.name,
+        if (reason != null) 'reason': reason,
+      }),
```

Comment:
We don't want a shared JsonSerializable for this? `TreeStatusChange` has repo/status/reason.

Reply:
Mental TODO, yeah this should just be using TreeStatusChange.toJson().